### ISSUE TITLE
CCM cannot load vSphere credentials through 'secretsDirectory' in yaml based config file.

### ISSUE THREAD

**Issue by wangyuan0916** [kubernetes/cloud-provider-vsphere#664]
### What happened?

CCM cannot load vSphere credentials through 'secretsDirectory' in yaml based config file.
Error logs:

> E1018 01:22:11.165416    9252 connection.go:63] Failed to create govmomi client. err: ServerFaultCode: Cannot complete login due to an incorrect user name or password.
> I1018 01:22:11.165448    9252 connectionmanager.go:152] Invalid credentials. Fetching credentials from secrets. vcServer=10.89.87.1 credentialHolder=Global
> W1018 01:22:11.165475    9252 credentialmanager.go:85] Cannot get secret  in namespace . error: "secret \"\" not found"
> E1018 01:22:11.165482    9252 credentialmanager.go:54] updateCredentialsMapK8s failed. err=secret "" not found
> W1018 01:22:11.165487    9252 credentialmanager.go:60] secret "" not found in namespace ""
> E1018 01:22:11.165492    9252 credentialmanager.go:75] credentials not found for server 10.89.87.1
> E1018 01:22:11.165497    9252 connectionmanager.go:162] Failed to get credentials from Secret Credential Manager with err:Credentials not found

 
I configured vsphere.conf:

global:
  port: 443
  insecureFlag: true
  secretsDirectory: /var/vcap/jobs/cloud-controller-manager/config/credentials/
vcenter:
  wy-Datacenter:
    server: **.**.**.**
    datacenters:
      - Datacenter
labels:
  region: wy-k8s-region
  zone: wy-k8s-zone

and put files as ‘**.**.**.**.password’ and ‘**.**.**.**.username’ with values as contents in the directory ‘/var/vcap/jobs/cloud-controller-manager/config/credentials/’.

### What did you expect to happen?

 ‘**.**.**.**.password’ and ‘**.**.**.**.username’  are loaded and parsed by CCM, and CCM can connect to vSphere.

### How can we reproduce it (as minimally and precisely as possible)?

Use secretsDirectory in yaml based config file

### Kubernetes version

1.24.0

### Cloud provider or hardware configuration

none

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Kernel (e.g. `uname -a`)

<details>

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### Others

<details>

</details>

**Comment by lubronzhan** [comment]
Do you need this to be included in 1.24 CPI?

**Comment by wangyuan0916** [comment]
I found it's possible to configure username and password in vsphere.conf directly, which is not mentioned in docs. So is it necessary to merge code?

**Comment by lubronzhan** [comment]
Yeah, still it's a bug. Also feel free to update the doc with your findings. I already cherry picked to 1.24. Will release a new one tomorrow

**Comment by lubronzhan** [comment]
Released in new 1.24.3
